subprojects/code-quality/src/integTest/groovy/org/gradle/api/plugins/quality/codenarc/CodeNarcGroovyVersionIntegrationTest.groovy

```diff
@@ -24,49 +24,65 @@ import org.gradle.util.internal.VersionNumber
 import static org.hamcrest.CoreMatchers.startsWith
 
 class CodeNarcGroovyVersionIntegrationTest extends AbstractIntegrationSpec implements CodeNarcTestFixture {
-    private static final STABLE_VERSION = CodeNarcPlugin.DEFAULT_CODENARC_VERSION
-    private static final STABLE_VERSION_WITH_GROOVY4_SUPPORT = "${STABLE_VERSION}-groovy-4.0"
-
     def setup() {
+        writeRuleFile()
+    }
+
+    void writeBuildFile(String groovyVersion, String codenarcVersion) {
+        String group = VersionNumber.parse(groovyVersion).major >= 4 ? "org.apache.groovy" : "org.codehaus.groovy"
         buildFile << """
             apply plugin: "groovy"
             apply plugin: "codenarc"
 
             ${mavenCentralRepository()}
 
             dependencies {
-                implementation(platform("org.apache.groovy:groovy-bom:${latestGroovy4Version}"))
-                implementation("org.apache.groovy:groovy")
+                implementation(platform("${group}:groovy-bom:${groovyVersion}"))
+                implementation("${group}:groovy")
 
-                codenarc("org.codenarc:CodeNarc:${STABLE_VERSION_WITH_GROOVY4_SUPPORT}")
-                codenarc(platform("org.apache.groovy:groovy-bom:${latestGroovy4Version}"))
+                codenarc("org.codenarc:CodeNarc:${codenarcVersion}")
```

Comment:
I think we could change this to setting the version as a tool version? e.g.

`codenarc.toolVersion = "${codenarcVersion}"`

Do we need to align the Groovy versions between CodeNarc and the project for CodeNarc to work? If not, we probably don't need to add the dependency to the groovy-bom in codenarc.

Reply:
I think I copied that from the original defect report, but you're correct, it's not necessary.  I've removed the bom dependency and used the tool version instead.